### DISCUSSION TITLE
SMART Train feed location is incorrect

### DISCUSSION THREAD

**Discussion by tatiang** [transitland/transitland#443] [Q&A] [answered]
I'm new to using the Transit Land API and I've been able to get results for the SMART Train feed here: https://www.transit.land/feeds/f-smart~ca~us. But the JSON data is referencing data from other countries such as Brazil and Italy rather than California. The SMART Train data should be for Sonoma and Marin counties in the United States.

Do you know why this might be? I'll post the url and partial JSON response below.

**https://transit.land/api/v2/rest/routes?served_by=f-smart~ca~us&apikey=[API KEY REMOVED]**

{"meta":{"after":154319,"next":"[https://transit.land/api/v2/rest/routes?after=154319\u0026apikey=[API KEY REMOVED]\u0026served_by=f-smart~ca~us](https://transit.land/api/v2/rest/routes?after=154319%5Cu0026apikey=[API KEY REMOVED]%5Cu0026served_by=f-smart~ca~us)"},"routes":[{"agency":{"agency_id":"1","agency_name":"Azienda Varesina Trasporti","id":4120,"onestop_id":"o-u0nkbh2-aziendavaresinatrasporti"},"continuous_drop_off":null,"continuous_pickup":null,"feed_version":{"feed":{"id":69,"onestop_id":"f-u0nkbh2-aziendavaresinatrasporti"},"fetched_at":"2016-03-14T20:27:49.554873Z","id":142,"sha1":"36f176346197292795b27e0ba455a2a3ca5d07ad"},"id":154308,"onestop_id":"r-u0nkbh-bf1","route_color":"07624A","route_desc":"La funicolare Vellone-Sacro Monte opera con frequenza ogni 10 minuti nei giorni feriali e festivi","route_id":"BF1","route_long_name":"Vellone - Sacro Monte","route_short_name":"BF1","route_sort_order":0,"route_text_color":"FFFFFF","route_type":7,"route_url":"[http://www.avtvarese.it/index.php?option=com_content\u0026view=category\u0026id=27\u0026Itemid=14](http://www.avtvarese.it/index.php?option=com_content%5Cu0026view=category%5Cu0026id=27%5Cu0026Itemid=14)"},{"agency":{"agency_id":null,"agency_name":"Bean Shuttle","id":4119,"onestop_id":"o-u2d-beanshuttle"},"continuous_drop_off":null,"continuous_pickup":null,"feed_version":{"feed":{"id":222,"onestop_id":"f-u2d-beanshuttle"},"

**Comment by drewda** [comment, marked as answer]
The routes endpoint does not have a "served_by" query parameter. Therefore your query was not filtering by anything and was just returning all routes from Transitland.

See https://www.transit.land/documentation/rest-api/routes for the allowed query parameters.

Try `https://transit.land/api/v2/rest/routes?operator_onestop_id=o-9qb-sonomamarinarearailtransit&apikey=`

**Comment by tatiang** [comment]
Drew,

Thank you for your help with that. The problem I'm having is that I don't
know how to get the stops for those routes. When I try this, the stops are
in other countries:

https://transit.land/api/v2/rest/stops?operator_onestop_id=o-9qb-sonomamarinarearailtransit&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

I did manage to get data from this but it takes a while and only lists 3 of
the 12 stations that exist:

https://transit.land/api/v2/rest/stops?served_by_onestop_ids=o-9qb-sonomamarinarearailtransit&lat=38.4405&lon=-122.7144&radius=10000&limit=100&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

I've tried a larger geographical area but it takes several minutes to
complete the call:

https://transit.land/api/v2/rest/stops?served_by_onestop_ids=o-9qb-sonomamarinarearailtransit&bbox=-122.70,37.80,-122.40,38.35&limit=100&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

Tatian

**Comment by tatiang** [comment]
I finally found a url that works to retrieve the stations correctly:

https://transit.land/api/v2/rest/routes/61941086?operator_onestop_id=o-9qb-sonomamarinarearailtransit&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

Tatian

**Comment by tatiang** [comment]
But then the route ID doesn't provide any stops:

https://transit.land/api/v2/rest/stops/*SA:SMART*
?operator_onestop_id=o-9qb-sonomamarinarearailtransit&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4
https://transit.land/api/v2/rest/stops/*1514161*
?operator_onestop_id=o-9qb-sonomamarinarearailtransit&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

Okay, this does seem to work for trips:

https://transit.land/api/v2/rest/routes/61941086/trips?operator_onestop_id=o-9qb-sonomamarinarearailtransit&service_date=2024-04-27&apikey=X2vfLI04BL3CTEtvz9uBqNPGrehkcvL4

Tatian
